momento-sdk/src/main/java/momento/sdk/CacheClientBuilder.java

```diff
@@ -3,28 +3,40 @@
 import java.time.Duration;
 import javax.annotation.Nonnull;
 import momento.sdk.auth.CredentialProvider;
+import momento.sdk.config.Configuration;
+import momento.sdk.config.transport.GrpcConfiguration;
+import momento.sdk.config.transport.TransportStrategy;
 
 /** Builder for {@link CacheClient} */
 public final class CacheClientBuilder {
 
   private final CredentialProvider credentialProvider;
+  private Configuration configuration;
   private final Duration itemDefaultTtl;
-  private Duration requestTimeout = null;
 
   CacheClientBuilder(
-      @Nonnull CredentialProvider credentialProvider, @Nonnull Duration itemDefaultTtl) {
+      @Nonnull CredentialProvider credentialProvider,
+      Configuration configuration,
```

Comment:
Should the configuration be @Nonnull too?

Reply:
Good catch.